### PR TITLE
Simplify package structure and decouple from Airbnb

### PR DESCRIPTION
**Pull request by jkrems**
* Drop dependency on airbnb base config
* Go back to one single package for all the things
* Use proper positive/negative tests
* Test pretty printing
* Add `prettier` for even more automated fixing
* Upgrade to `eslint@4`

Still pending:

- [x] Port all node 6 rules
- [x] Port all node 4 rules
- [x] Create node 8 rules
- [x] Port all ES5 rules
- [x] ~~Port CSS formatting to prettier~~
- [x] Update README/doc files
- [x] Set up `nlm` (again) and make it publishable.

Rules that didn't make the cut (selection):

- Disallow `x++` and mixed operators. No `--fix` yet.
- Global strict is only a warning, pending a `--fix` solution.

I tried to be conservative about adding back rules in general. I think we cover all the actual bug prevention now and `prettier` should cover formatting.

### CODE REVIEW

**Comment by jkrems** [comment]
I think this is ready for a fist review.

File: package.json [hunk 1]
```diff
@@ -14,28 +15,35 @@
...
+        "lib"
+      ]
+    }
+  },
   "dependencies": {
-    "coffeelint-config-groupon": "file:./packages/coffeelint-config-groupon",
     "coffeelint-forbidden-keywords": "~0.1.1",
     "coffeelint-no-mocha-only": "^1.0.0",
     "coffeelint-use-strict": "^1.0.0",
     "coffeescope2": "~0.4.2",
```
Comment (dbushong) [inline package.json]: Should these be peerDeps instead of direct deps?

File: package.json [hunk 2]
```diff
@@ -1,9 +1,10 @@
 {
-  "name": "groupon-styleguide",
-  "version": "3.2.0",
-  "private": true,
+  "name": "eslint-config-groupon",
```
Comment (dbushong) [inline package.json]: Does this mean for coffeelint it'll be `"extends": "eslint-config-groupon/coffee"` (similarly for stylus/css?)  should we choose a new module name?

**Comment by jkrems** [comment]
Added commits to enable `nlm` and removed the WIP prefix.

**Comment by dbushong** [comment]
Shouldn't this be semver-major?

File: package.json (hunk 2 above)
Comment (jkrems) [inline package.json]: I think `eslint` is the only one that cares about the naming. In theory we could just go for `groupon-code-style` or similar. IIRC `eslint` also accepts full package names..?

File: package.json (hunk 1 above)
Comment (jkrems) [inline package.json]: Heh, good point. Yeah, I think so. Blindly left these here based on the previous top-level state which is clearly not correct.

**Comment by jkrems** [comment]
Re major: Yep! Let me reword the first commit to include the marker.

File: README.md [hunk 3]
```diff
@@ -39,82 +26,25 @@ This ensures that:
 
 Fortunately there already is a great and well-documented style guide for JavaScript over at [airbnb/javascript](https://github.com/airbnb/javascript).
 The short answer is: We stick to that.
```
Comment (dbushong) [inline README.md]: Is this even true or worth saying anymore?

Comment (jkrems) [reply README.md]: Yeah, we should reword this. Maybe turn it into an acknowledgement of where we started.

**Pull request merged**